playbulb-candle/playbulbCandle.js

```diff
@@ -114,6 +116,20 @@
         this._isEffectSet = true;
       });
     }
+    startBlowNotifications(listener) {
+      let characteristic = this._characteristics.get(CANDLE_BLOW_NOTIFICATIONS_UUID);
+      return characteristic.startNotifications()
+      .then(_ => {
+        characteristic.addEventListener('characteristicvaluechanged', listener);
+      });
+    }
+    stopBlowNotifications(listener) {
+      let characteristic = this._characteristics.get(CANDLE_BLOW_NOTIFICATIONS_UUID);
+      return characteristic.stopNotifications()
+      .then(_ => {
+        characteristic.removeEventListener('characteristicvaluechanged', listener);
```

Comment:
If `stopNotifications()` rejects, removing it after will mean the characteristic's still notifying, and the listener's still attached. Removing it before will mean the characteristic's still notifying, and the listener's not attached. Removing it after means that on rejection you just wouldn't call `startNotifications()` if you start wanting notifications again; you'd use the old listener. Removing it before means you'd need to call `startNotifications()` to re-attach the listener. 'course, if `stopNotifications()` rejects, something's pretty wrong, and you may not be getting notifications anyway.

I don't really know which way this argues.